packages/react-keytips/src/hooks/useHotkeys.ts

```diff
@@ -79,15 +89,37 @@ export const useHotkeys = (hotkeys: Hotkey[], target?: Document) => {
               ev.stopPropagation();
             }
 
-            handler(event);
+            // if options.delay is > 0, it will be needed to hold the sequence
+            // to fire the handler
+            if (options.delay && options.delay > 0) {
+              clearDelayTimeout();
+              activeKeys.current.add(ev.key);
+              setDelayTimeout(() => {
+                if (activeKeys.current.has(ev.key)) {
+                  handler(event);
+                }
+              }, options.delay);
+            } else {
+              handler(event);
+            }
           }
         }
       );
     };
 
-    doc?.addEventListener('keydown', listener);
+    const onKeyUp = (ev: KeyboardEvent) => {
+      activeKeys.current.delete(ev.key);
+
+      if (activeKeys.current.size === 0) {
+        clearDelayTimeout();
+      }
+    };
+
+    doc?.addEventListener('keydown', onKeyDown);
+    doc?.addEventListener('keyup', onKeyUp);
     return () => {
```

Comment:
Do we need to invoke the `clearDelayTimeout` function during the unmount phase?

Reply:
current keyUp ensures, that it will be called, but in general -> it's a good practice to clean all pending timeouts on umount. If this is what u mean, I can add it there explicitly